server/chat-plugins/helptickets.ts

```diff
@@ -1380,3 +1390,24 @@ export const punishmentfilter: Chat.PunishmentFilter = (user, punishment) => {
 	const ticket = helpRoom.game as HelpTicket;
 	ticket.close('ticketban');
 };
+
+// save ticket state before hotpatch
+export const destroy = () => {
+	for (const room of Rooms.rooms.values()) {
+		if (!room.settings.isHelp || !room.game) continue;
+		const game: HelpTicket = room.getGame(HelpTicket)!;
+		const entry: TicketState = {
+			...game.ticket,
+			resolution: game.resolution,
+			result: game.result || undefined,
+			claimQueue: game.claimQueue,
+			involvedStaff: [...game.involvedStaff],
+			createTime: game.createTime,
+			emptyRoom: game.emptyRoom,
+			firstClaimTime: game.firstClaimTime,
+			unclaimedTime: game.unclaimedTime,
+		};
+		tickets[game.ticket.userid] = entry;
+	}
+	writeTickets();
+};
```

Comment:
I think it's much better to track these sorts of things in `game.ticket` only, instead of having to write a `destroy` handler.

Reply:
your code also desyncs `tickets` and `game.ticket` which causes a bug on restarts where it'll have old data if you hotpatch chat.

Reply:
We talked about this in Discord: the main problem is that some properties are inside `game.ticket` and some are directly in `game`. They should either all be moved to `game.ticket` or all moved to `game` (and `.ticket` would be autogenerated from `game` when `writeTickets` is called). Either approach would remove the need for this `destroy` function.